python/ribasim/ribasim/config.py

```diff
@@ -44,9 +44,10 @@ class Compression(str, Enum):
 
 
 class Results(BaseModel):
-    basin: Path = Path("results/basin.arrow")
-    flow: Path = Path("results/flow.arrow")
-    control: Path = Path("results/control.arrow")
+    basin: Path = Field(default=Path("basin.arrow"), exclude=True, repr=False)
+    flow: Path = Field(default=Path("flow.arrow"), exclude=True, repr=False)
+    control: Path = Field(default=Path("control.arrow"), exclude=True, repr=False)
+    allocation: Path = Field(default=Path("allocation.arrow"), exclude=True, repr=False)
```

Comment:
These need to be removed too

Reply:
I've excluded them from any exports. They are here for when we need to parse the output files.

Reply:
Just an FYI: I just found out that TOML hates these Path objects, and complains that it cannot serialize them.

However, due to the exclude_unset logic, they aren't included unless you set them yourself. For extra confusion, if you set the path (in current main) as a string, pydantic will coerce it to a Path. Then pydantic detects it has been set, and then TOML complains that it cannot serialize it.

To be honest, together with issue #811, I'm getting really radicalized against exclude_unset.

Reply:
That's why there's 

    @field_serializer("input_dir", "results_dir") 
      def serialize_path(self, path: Path) -> str: 
          return str(path)


Right? Which fields are you trying to set?

I don't like the unset behaviour either, but that should be a separate issue/discussion.